onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java

```diff
@@ -831,7 +890,14 @@ void runSimulation(int expectedGetReadyIndex, int expectedPullCordIndex) {
                 } catch (Exception e) {
                     Log.e(TAG, e.getMessage());
                 }
-                provider.locationUpdated(l);
+
```

Comment:
I believe above line 886 when this 2nd `for` loop starts, you'll need to re-initialize mPrevLocation` to `mLocations[expectedGetReadyIndex - 1]`.

The first `for` loop ends with `i = expectedGetReadyIndex`, and the second for loop starts with `i = expectedGetReadyIndex`, so starting the 2nd loop `mPrevLocation` and  `mLocations[i]` would both be pointing to `mLocations[expectedGetReadyIndex]`, which will then always be skipped.

Please take a closer look at this to make sure I'm reading the code right.